wrapper/src/main/java/software/amazon/jdbc/dialect/DialectManager.java

```diff
@@ -0,0 +1,103 @@
+/*
+ * Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License").
+ * You may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package software.amazon.jdbc.dialect;
+
+import java.sql.SQLException;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.Properties;
+import org.checkerframework.checker.nullness.qual.NonNull;
+import software.amazon.jdbc.AwsWrapperProperty;
+import software.amazon.jdbc.HostSpec;
+import software.amazon.jdbc.util.ConnectionUrlParser;
+import software.amazon.jdbc.util.Messages;
+import software.amazon.jdbc.util.RdsUtils;
+import software.amazon.jdbc.util.StringUtils;
+import software.amazon.jdbc.util.Utils;
+
+public class DialectManager implements DialectProvider {
+
+  protected static DialectProvider customDialectProvider;
+
+  public static final AwsWrapperProperty DIALECT = new AwsWrapperProperty(
+      "wrapperDialect", "",
+      "A unique identifier for the supported database dialect. ");
+
+  protected static final Map<String, Dialect> knownDialectsByCode =
+      new HashMap<String, Dialect>() {
+        {
+          put("mysql", new MysqlDialect());
+          put("pg", new PgDialect());
+          put("mariadb", new MariaDbDialect());
+          put("aurora-mysql", new AuroraMysqlDialect());
+          put("aurora-pg", new AuroraPgDialect());
+        }
+      };
+
+  private final RdsUtils rdsHelper = new RdsUtils();
+  private final ConnectionUrlParser connectionUrlParser = new ConnectionUrlParser();
+
+  public static void setCustomDialectProvider(final @NonNull DialectProvider dialectProvider) {
+    customDialectProvider = dialectProvider;
+  }
+
+  public static void resetCustomDialectProvider() {
+    customDialectProvider = null;
+  }
+
+  public Dialect getDialect(String driverProtocol, String url, Properties props)
+      throws SQLException {
+
+    String userDialectSetting = DIALECT.getString(props);
+    Dialect userDialect = knownDialectsByCode.get(userDialectSetting);
```

Comment:
Please final these variables. You can use IntelliJ for this. `ctrl+shift+alt+i`, search for `local variable of parameter can be final`, choose the files you want to modify and click `fix all`